### PR TITLE
rg-156: Add error for registration page (email taken)

### PR DESCRIPTION
**Pull request by Icerock10**
image

### CODE REVIEW

**Review by ValentinaTenus — APPROVED** [review]

File: shared/src/enums/server-error-type.enum.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 enum ServerErrorType {
     COMMON = 'COMMON',
     VALIDATION = 'VALIDATION',
+    EMAIL = 'email',
```
Comment (MaxPoplavskiy) [inline shared/src/enums/server-error-type.enum.ts:4]: Consider using UPPERCASE

**Review by Valentyn13 — APPROVED** [review]

**Review by MaxPoplavskiy — CHANGES_REQUESTED** [review]

**Review by MaxPoplavskiy — APPROVED** [review]

File: frontend/src/bundles/preview/preview.tsx [hunk 2]
```diff
@@ -246,9 +246,7 @@ const PreviewPage: React.FC = () => {
                 <li className={styles.item}>
                     <Templates />
                 </li>
-                <li className={styles.auth}>
-                    <Auth />
-                </li>
+                <li className={styles.auth}></li>
```
Comment (jurok3x) [inline frontend/src/bundles/preview/preview.tsx:249]: empty list item?

**Review by jurok3x — APPROVED** [review]

**Review by honkster-nazarii — APPROVED** [review]

File: frontend/src/bundles/common/hooks/use-form-error/use-form-error.hook.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+type ReturnValue = {
+    dataStatus: DataStatus;
+};
+
+const useFormError = ({ setError }: UseFormErrorPayload): ReturnValue => {
+    const { error, dataStatus } = useAppSelector((state) => state.auth);
+
+    useEffect(() => {
+        if (error && dataStatus === DataStatus.REJECTED) {
```
Comment (Vitaliy-Kalenichenko) [inline frontend/src/bundles/common/hooks/use-form-error/use-form-error.hook.ts:25]: Is there a case where it's error and dataStatus is not rejected?

Comment (Icerock10) [reply frontend/src/bundles/common/hooks/use-form-error/use-form-error.hook.ts]: yes, data status check is redundant)

**Review by Valentyn13 — APPROVED** [review]

**Pull request merged**